node/core/approval-voting/src/lib.rs

```diff
@@ -378,6 +378,26 @@ impl ApprovalVotingSubsystem {
 	}
 }
 
+// Checks and logs approval vote db state. It is perfectly normal to start with an
+// empty approval vote DB if we changed DB type or the node will sync from scratch.
+fn db_sanity_check(db: Arc<dyn Database>, config: DatabaseConfig) -> SubsystemResult<()> {
+	let backend = DbBackend::new(db, config);
+	let backend = OverlayedBackend::new(&backend);
+	let all_blocks = backend.load_all_blocks()?;
```

Comment:
Do we really have to load all blocks to tell whether there are any at all?

Reply:
There is no way to iterate if the column is not ordered. I assume you are thinking this is going to take a lot of time, but this also happens during normal operation of the subsystem. I could make this less expensive than it is, but my thinking was it would be great to also debug log how many blocks are there if its not empty

Reply:
If paritydb supports getting number of records in a column, it might be simplier in future. There was a PR with a discussion on that topic: https://github.com/paritytech/parity-db/pull/94

Reply:
I talked with @cheme about this and it seemed that it was not a good idea to have an API for that purpose only. But yeah, this would indeed make this easier as we could do it when opening the db instead of subsystem level.